### PR TITLE
Route groupings

### PR DESCRIPTION
**Pull request by lemald**
#### Summary of changes
**Asana Ticket:** [[2] Add route groupings to analyzer](https://app.asana.com/0/584764604969369/1120741346356341/f)

Updated `filter_by_route` to call a utility function that converts the text from the parameters to a list of route IDs. For most cases this list will contain only the one route.

#### Reviewer Checklist
- [ ] Meets ticket's acceptance criteria
- [ ] Any new or changed functions have typespecs
- [ ] Tests were added for any new functionality (don't just rely on Codecov)

### CODE REVIEW

**Comment by losvedir** [comment]
So it seems the idea here is there's a "magic word" that our backend knows how to translate into certain sets of routes.

An alternative approach might be to teach the app to allow `route_id=` to be enumerated with commas (a la the API), and have the view template simply list the route set in the options: `[{"Green-All", "Green-B,Green-C,Green-D,Green-E"}, ...]`.

The advantage would be that we could then query sets of routes without special backend support. For routes I'm not sure it matters much, but for stop_ids I could see it being more relevant.

That was sort of how I was envisioning it, but I see the asana ticket isn't very explicit about it. Given that it's already implemented this way, if it's a lot of work it might not be worth changing it now. But I do think stop_ids should work like that, at least.

**Comment by lemald** [comment]
Hmm, it might not actually be too much work to re-do it that way, and now that you mention it I definitely agree that that's how stop IDs should work. I'll consider it.

File: lib/prediction_analyzer/utilities.ex [hunk 1]
```diff
@@ -91,4 +91,13 @@ defmodule PredictionAnalyzer.Utilities do
   @spec string_to_mode(String.t()) :: :subway | :commuter_rail
   def string_to_mode("commuter_rail"), do: :commuter_rail
   def string_to_mode(_), do: :subway
+
+  @spec route_param_to_list(String.t()) :: [String.t()]
```
Comment (losvedir) [inline lib/prediction_analyzer/utilities.ex]: I don't think this function is used anymore?

Comment (lemald) [reply lib/prediction_analyzer/utilities.ex]: Ah, that is correct, it is not used.

**Pull request merged**